msbuild/Xamarin.MacDev.Tasks.Core/Tasks/IBToolTaskBase.cs

```diff
@@ -218,21 +221,24 @@ public override bool Execute ()
 					continue;
 				}
 
-				if (UseCompilationDirectory) {
-					rpath = Path.Combine (ibtoolOutputDir, Path.GetDirectoryName (bundleName));
-					output = new TaskItem (rpath);
-					outputDir = rpath;
+				rpath = Path.Combine (ibtoolOutputDir, bundleName);
+				outputDir = Path.GetDirectoryName (rpath);
+				output = new TaskItem (rpath);
 
-					output.SetMetadata ("LogicalName", Path.GetDirectoryName (bundleName));
-				} else {
-					rpath = Path.Combine (ibtoolOutputDir, bundleName);
-					outputDir = Path.GetDirectoryName (rpath);
-					output = new TaskItem (rpath);
+				output.SetMetadata ("LogicalName", bundleName);
+				output.SetMetadata ("Optimize", "false");
 
-					output.SetMetadata ("LogicalName", bundleName);
+				if (Path.GetExtension (bundleName) != ".plist") {
+					// Don't include Watch storyboards that got compiled to plists
+					compiled.Add (output);
 				}
 
```

Comment:
Style: Extra {} (I think, I'm not great with the mono style, if not, sorry for the noise).